### ISSUE TITLE
colmap stage 0 failed

### ISSUE THREAD

**Issue by r530044129** [ashawkey/nerf2mesh#5]
![image](https://user-images.githubusercontent.com/22882223/224002406-3988f5f6-e6e5-4876-8b53-775a3eb5fdc0.png)
![image](https://user-images.githubusercontent.com/22882223/224002469-ef1a115c-a945-476f-a498-6e7937fdc9ac.png)

I just follow your tutorial, which use garden to generate, i not sure why i failed

the stage 0 of nerf is success

**Comment by r530044129** [comment]
I use ubuntu 18, python 3.8, torch 1.13.1, cuda 11.6

**Comment by ashawkey** [comment]
@r530044129 Hi, have you modified the code? NVdiffrast is not supposed to run in CUDA context in non-gui mode as there are many limitations, like this one. 
You could make H/W divisible by 8 before passing it to ``mark_unseen_triangles` if you have to use CUDA context, but it will still error out in texture exportation (unless you reduce texture resolution to 2048 and ssaa to 1, since the max resolution CUDA context support is 2048).

**Comment by r530044129** [comment]
@ashawkey yeah, I forget tell you that I do have changed code, in mark_unseen_triangles(),I change OpenGL to Cuda, around line 818.
Because When I use default code, there is an Aborted after Saving Mesh, see this picture
[F glutil.cpp:338] eglInitialize() failed
![image](https://user-images.githubusercontent.com/22882223/224079403-fac5a17f-d324-40db-abe3-7a472b100eec.png)
And after some google search, someone suggest maybe I could use cuda to replace Opengl, but it's seems not work too, lol
Hope you can help me to figure out the Aborted Problem, thanks!

**Comment by ashawkey** [comment]
This indicates the OpenGL installation is somewhat wrong. You may refer to this https://github.com/sicxu/Deep3DFaceRecon_pytorch/issues/2.
